src/Details.js

```diff
@@ -1,28 +1,28 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
+import getCountries from './countries-get'
 
 export default class Details extends React.Component {
 
     constructor() {
         super();
         this.state = {
-            welcomeMessage: 'Welcome to the details page, WIP :)'
+            country: {}
         };
     }
 
     componentDidMount() {
-        setTimeout(() => {
-            this.setState({
-                welcomeMessage: 'The best is yet to come!'
-            });
-        }, 3000);
+        let countryId = this.props.match.params.countryId;
+        let country = getCountries()
+            .find((country) => country.id === countryId);
+        this.setState({ country });
     }
 
     render() {
         return (
-        <div><h1>{this.state.welcomeMessage}</h1>
-            <Link to='/'>Back to home page</Link>
-        </div>
+            <div><h1>{this.state.country.name}</h1>
```

Comment:
I think its better if you nested the items rather than put it together in the same line, i.e
<div><h1>               <div>
     <link...      ---->      <h1>
                                      <link>
because <h1> and <link> are both 'childrens' inside the <div> 'parent' box.